src/main/java/com/metamx/emitter/service/ServiceMetricEvent.java

```diff
@@ -250,6 +256,12 @@ public Builder setUser10(String x)
       return this;
     }
 
+    public Builder setDimension(String dim, String value)
+    {
+      userDims.put(dim, value);
```

Comment:
This class isn't meant to be used only as input to druid, it could also be logged to a file or posted to a web service. So I don't think it necessarily has to deal with nulls/empties the same way that druid does. If it does end up getting used as input to druid, then druid is free to interpret the null values however it wants.